components/pkg-export-docker/src/build.rs

```diff
@@ -328,10 +339,10 @@ impl<'a> BuildSpec<'a> {
                               -> Result<FullyQualifiedPackageIdent> {
         self.install(ui,
                      ident_or_archive,
-                     self.url,
+                     &self.url,
                      &self.channel,
                      fs_root_path,
-                     self.auth)
+                     self.auth.as_ref().map(String::as_str)) //eww
```

Comment:
Using [`as_deref`](https://doc.rust-lang.org/stable/std/option/enum.Option.html#method.as_deref) makes this cleaner.

Reply:
Ah, yes :+1:

Reply:
Fixed in https://github.com/habitat-sh/habitat/pull/7660/commits/da8926ab7034998abd66c172fd0e66a062a0f4d3